src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -126,6 +126,58 @@ public void doReserve(StaplerRequest req, StaplerResponse rsp)
 		rsp.forwardToPreviousPage(req);
 	}
 
+	public void doReassign(StaplerRequest req, StaplerResponse rsp)
+		throws IOException, ServletException {
+		Jenkins.getInstance().checkPermission(RESERVE);
```

Comment:
This is a permission check.

Reply:
Looking at the source, `doSteal()` started its life as a verbatim copy of `doReserve()`, and `doReassign()` as a modified copy of `doUnreserve()` and so (perhaps mindlessly) carry the same baggage.

Looking at existing permissions nearby, maybe an additional check for "UNLOCK" makes sense (as in `doUnlock()` and `doReset()` functions/buttons).

Reply:
Hopefully got the question better now, I'm not too slow =D
Added a commit to define new permission "STEAL" for both of these cases (steal from a locking job, reassign from a reserving user) and changed the UI and java checks into using that.